ocean_rough.F90

```diff
@@ -287,7 +326,155 @@ subroutine ocean_rough_init
 
  end subroutine ocean_rough_init
 
+
 !#######################################################################
 
+ subroutine cal_z0_hwrf17 (uref,z0)
+
+! This subroutine is orginally from HWRF model (2017 version, znot_m_v6)
+! Implemented here by Kun Gao & Baoqiang Xiang
+
+! Calculate areodynamical roughness over water with input 10-m wind
+! For low-to-moderate winds, try to match the Cd-U10 relationship from COARE V3.5 (Edson et al. 2013)
+! For high winds, try to fit available observational data
+!
+! uref(m/s)   :   wind speed at 10-m height
+! z0 (meter)  :   areodynamical roughness scale over water
+!
+
+    real, intent(in) :: uref
+    real, intent(out):: z0
+    real  :: p13, p12, p11, p10
+    real  :: p25, p24, p23, p22, p21, p20
+    real  :: p35, p34, p33, p32, p31, p30
+    real  :: p40
+
+    p13 = -1.296521881682694e-02
+    p12 =  2.855780863283819e-01
+    p11 = -1.597898515251717e+00
+    p10 = -8.396975715683501e+00
+
+    p25 =  3.790846746036765e-10
+    p24 =  3.281964357650687e-09
+    p23 =  1.962282433562894e-07
+    p22 = -1.240239171056262e-06
+    p21 =  1.739759082358234e-07
+    p20 =  2.147264020369413e-05
+
+    p35 =  1.840430200185075e-07
+    p34 = -2.793849676757154e-05
+    p33 =  1.735308193700643e-03
+    p32 = -6.139315534216305e-02
+    p31 =  1.255457892775006e+00
+    p30 = -1.663993561652530e+01
+
+    p40 =  4.579369142033410e-04
+
+    if (uref >= 0.0 .and.  uref <= 6.5 ) then
+      z0 = exp( p10 + p11*uref + p12*uref**2 + p13*uref**3)
+    elseif (uref > 6.5 .and. uref <= 15.7) then
+      z0 = p25*uref**5 + p24*uref**4 + p23*uref**3 + p22*uref**2 + p21*uref + p20
+    elseif (uref > 15.7 .and. uref <= 53.0) then
+      z0 = exp( p35*uref**5 + p34*uref**4 + p33*uref**3 + p32*uref**2 + p31*uref + p30 )
+    elseif ( uref > 53.0) then
+      z0 = p40
+    else
+      print*, 'Wrong input uref value:',uref
+    endif
+!
+    z0 = max(z0, roughness_min)
```

Comment:
Thanks, Kun. If we bring this into computer_ocean_roughness, it may have different values when we update this through ocean_rough.F90 compared to the one in surface_flux.F90 (although it may not be effective at all). I did a run to check and the model knows the value of roughness_min when we call it in surface_flux.F90

Reply:
I just feel if a lower limit is to be applied on z0, this lower bound should be more explicitly defined.

Reply:
This lower limit has no impact and I just removed it.